File: Python/Ejercicios Xmas/CuentaCorriente.py
from DNIChecker import *

class cuentaCorriente:
    def __init__(self, nombre, apellidos, direccion, telefono, dni, saldo=None):
        self.nombre = nombre
        self.apellidos = apellidos
        self.direccion = direccion
        self.telefono = telefono
        self.dni = dni
        self.saldo = saldo

    def setNombre(self, nombre):
        self.nombre = nombre

    def getNombre(self):
        self.nombre

    def setApellidos(self, apellidos):
        self.apellidos = apellidos

    def getApellidos(self):
        self.apellidos

    def setDireccion(self, direccion):
        self.direccion = direccion

    def getDireccion(self):
        self.direccion

    def setTelefono(self, telefono):
        self.telefono = telefono

    def getTelefono(self):
        self.telefono

    def setDNI(self, dni):
        self.dni = dni

    def getDNI(self):
        self.dni

    def setSaldo(self, saldo):
        self.saldo=int(saldo)

    def getSaldo(self):
        return self.saldo

    def retirarDinero(self, cantidad):
        self.setSaldo(self.getSaldo() - int(cantidad))

    def ingresarDinero(self, cantidad):
        self.setSaldo(self.getSaldo() + int(cantidad))    

    def consultarCuenta(self):
        usuario = "Usuario: " + str(self.nombre) + " " + str(self.apellidos) + "\n"
        identificacion = "Con DNI: " + str(self.nif)
        numeroDeTelefono = "Numero de telefono: " + str(self.telefono)
        domicilio = "Domicilio: " + str(self.direccion)

        print (usuario, " identificacion" + "\n", numeroDeTelefono + "\n", domicilio)

File: Java/POO_Navidad/src/org/foobarspam/POO/cuentacorriente/CuentaCorriente.java
package org.foobarspam.POO.cuentacorriente;

import org.foobarspam.POO.DNI.*;

public class CuentaCorriente {
	// Propiedades
	private String nombre;
	private String apellidos;
	private String direccion;
	private String telefono;
	private DNI nif;
	private Double saldo;
	
	// Constructores
	public CuentaCorriente() {
		
	}
	
	public CuentaCorriente(String nombre, String apellidos, String direccion, String telefono, DNI nif, double saldo){
		this.nombre = nombre;
		this.apellidos = apellidos;
		this.direccion = direccion;
		this.telefono = telefono;
		this.nif = nif;
		this.saldo = saldo;
	}

	// Setters y getters
	public Double getSaldo() {
		return saldo;
	}

	public void setSaldo(Double saldo) {
		this.saldo = saldo;
	}

	public String getNombre() {
		return nombre;
	}

	public String getApellidos() {
		return apellidos;
	}

	public String getDireccion() {
		return direccion;
	}

	public String getTelefono() {
		return telefono;
	}

	public DNI getNif() {
		return nif;
	}
	
	// Métodos
	public Double retirarDinero(Double dinero){
		setSaldo(getSaldo() - dinero);
		return getSaldo();
	}
	
	public Double ingresarDinero(Double dinero){
		setSaldo(getSaldo() + dinero);
		return getSaldo();
	}
	
	public void consultarCuenta(){
		System.out.println("La cuenta a nombre de: " + getNombre() + " " + getApellidos());
		System.out.println("Con la dirección: " + getDireccion());
		System.out.println("Con el teléfono: " + getTelefono());
		System.out.println("Y NIF: " + getNif().getCompleto());
		System.out.println("Dispone de " + getSaldo() + "€ en su cuenta corriente");
	}
	
	public Boolean saldoNegativo(){
		if (getSaldo() < 0){
			return true;
		} else {
			return false;
		}
	}
}
